src/NextLink.js

```diff
@@ -0,0 +1,41 @@
+import React, { forwardRef } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import Link from 'next/link';
+
+export const NextLink = forwardRef(({ children, className, href, newTab, prefetch, as, scroll, external, ...props }, ref) => {
+    const Anchor = (
+        <a
+            ref={ ref }
+            className={ classNames('moxy-next-link', className) }
+            target={ newTab ? '_blank' : undefined }
+            rel={ newTab ? 'noopener noreferrer' : undefined }
+            href={ external ? href : undefined }
+            { ...props }>
+            { children }
+        </a>
+    );
+
+    if (external) {
+        return Anchor;
+    }
+
+    return (
+        <Link href={ href } prefetch={ prefetch && undefined } as={ as } scroll={ scroll } passHref>
```

Comment:
what is `passHref`?

Reply:
It's [a prop from next/link](https://nextjs.org/docs/api-reference/next/link#if-the-child-is-a-custom-component-that-wraps-an-a-tag):

> If the child of `Link` is a custom component that wraps an `<a>` tag, you must add `passHref` to `Link`. Without this, the `<a>` tag will not have the `href` attribute, which might hurt your site’s SEO.